plat/common/arm/spinlock64.S

```diff
@@ -0,0 +1,71 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
```

Comment:
Would lib/uklock/ be a more suitable place for `spinlocks.h` and its implementation?

Reply:
We also thought about this and we had a 50/50-feeling when deciding where to put it. At the moment all primitives in `uklock` interact with the scheduler for thread blocking, so `uklock` has a dependency on it. We therefore thought that it might be better to see the spinlock as some kind of basic data type/primitive and allow it in the architectural headers (hence the change in the naming to `__spinlock`).

Of course, we could also make the dependency in `uklock` more differentiated so that the scheduler is only involved if `uk_mutex` or `uk_semaphore` are added per config. But that would create a hard dependency of most low-level SMP-ready code (e.g., platforms) on `uklock`.

What are your thoughts on this?

Reply:
I think in that case uklock would be a dependency of SMP with our without a scheduler, but since uklock is an internal library does it make a difference?